scripts/pre_commit_linter.py

```diff
@@ -102,12 +102,47 @@
         'regexp': r"\b(ddescribe|fdescribe)\(",
         'message': "In tests, please use 'describe' instead of 'ddescribe'"
                    "or 'fdescribe'",
-        'excluded_files': ()
+        'excluded_files': (),
+        'excluded_dirs': ()
     },
     {
         'regexp': r"\b(iit|fit)\(",
         'message': "In tests, please use 'it' instead of 'iit' or 'fit'",
-        'excluded_files': ()
+        'excluded_files': (),
+        'excluded_dirs': ()
+    },
+    {
+        'regexp': r"templateUrl: \'",
+        'message': "The directives must be directly referenced.",
+        'excluded_files': (
+            'core/templates/dev/head/pages/exploration_editor/'
+            'editor_tab/StateResponses.js',
+            'core/templates/dev/head/pages/exploration_player/'
+            'FeedbackPopupDirective.js'
+            ),
+        'excluded_dirs': (
+            'extensions/')
+    }
+]
+
+BAD_PATTERNS_HTML_REGEXP = [
+    {
+        'regexp': r"text\/ng-template",
+        'message': "The directives must be directly referenced.",
+        'excluded_files': (
+            'core/templates/dev/head/pages/exploration_editor/'
+            'editor_tab/state_editor_responses.html',
+            'core/templates/dev/head/pages/exploration_player/'
+            'conversation_skin_directive.html',
+            'core/templates/dev/head/pages/exploration_player/'
+            'feedback_popup_container_directive.html',
+            'core/templates/dev/head/pages/exploration_player/'
+            'learner_local_nav.html',
+            'core/templates/dev/head/pages/exploration_player/'
```

Comment:
Why exclude this one?

Reply:
it has `popover/answer` which has same problem as `popover/feedback`